vendor/dragino/lwl03a.js

```diff
@@ -0,0 +1,162 @@
+  return c_Time;
+}
+
+
+function Decoder(bytes, port) {
+  if(port==0x03)
+  {
+    for(var i=0;i<bytes.length;i=i+11)
+    {
+      var data= datalog(i,bytes);
+      if(i=='0')
+        data_sum=data;
+      else
+        data_sum+=data;
+    }
+    return{
+    DATALOG:data_sum
+    };
+  }
+  else if(port==0x04)
+  {
+    var tdc= bytes[0]<<16 | bytes[1]<<8 | bytes[2];
+    var disalarm= bytes[3]&0x01;
+    var keep_status= bytes[4]&0x01;
+    var keep_time= bytes[5]<<8 | bytes[6];
+    var leak_alarm_time= bytes[7];
+  
+    return {
+    TDC:tdc,
+    DISALARM:disalarm,
+    KEEP_STATUS:keep_status,
+    KEEP_TIME:keep_time,
+    LEAK_ALARM_TIME:leak_alarm_time
+    };
+  }
+  else if(port==0x05)
+  {
+    var sub_band;
+    var freq_band;
+    var sensor;
+   
+    if(bytes[0]==0x14)
+      sensor= "LWL03A";
+     
+    if(bytes[4]==0xff)
+      sub_band="NULL";
+    else
+      sub_band=bytes[4];
+   
+    if(bytes[3]==0x01)
+      freq_band="EU868";
+    else if(bytes[3]==0x02)
+      freq_band="US915";
+    else if(bytes[3]==0x03)
+      freq_band="IN865";
+    else if(bytes[3]==0x04)
+      freq_band="AU915";
+    else if(bytes[3]==0x05)
+      freq_band="KZ865";
+    else if(bytes[3]==0x06)
+      freq_band="RU864";
+    else if(bytes[3]==0x07)
+      freq_band="AS923";
+    else if(bytes[3]==0x08)
+      freq_band="AS923_1";
+    else if(bytes[3]==0x09)
+      freq_band="AS923_2";
+    else if(bytes[3]==0x0A)
+      freq_band="AS923_3";
+    else if(bytes[3]==0x0B)
+      freq_band="CN470";
+    else if(bytes[3]==0x0C)
+      freq_band="EU433";
+    else if(bytes[3]==0x0D)
+      freq_band="KR920";
+    else if(bytes[3]==0x0E)
+      freq_band="MA869";
+     
+    var firm_ver= (bytes[1]&0x0f)+'.'+(bytes[2]>>4&0x0f)+'.'+(bytes[2]&0x0f);
+    var bat= (bytes[5]<<8 | bytes[6])/1000;
+   
+    return {
+    SENSOR_MODEL:sensor,
+    FIRMWARE_VERSION:firm_ver,
+    FREQUENCY_BAND:freq_band,
+    SUB_BAND:sub_band,
+    BAT:bat,
+    };
+  }
+  else
+  {
+    var tdc_interval=(bytes[0]&0x04)?"YES":"NO";
+    var alarm=(bytes[0]&0x02)?"TRUE":"FALSE";
+    var leak_status=(bytes[0]&0x01)?"LEAK":"NO LEAK"; 
+    var leak_times=bytes[1]<<16 | bytes[2]<<8 | bytes[3];
+    var leak_duration=bytes[4]<<16 | bytes[5]<<8 | bytes[6];
+    var data_time= getMyDate((bytes[7]<<24 | bytes[8]<<16 | bytes[9]<<8 | bytes[10]).toString(10));
+   
+    if(bytes.length==11)
+    {
+      return {
+      TDC:tdc_interval,
+      ALARM:alarm,
+      WATER_LEAK_STATUS:leak_status,
+      WATER_LEAK_TIMES:leak_times,
+      LAST_WATER_LEAK_DURATION:leak_duration,
+      TIME:data_time
+      };
+    }
+  }
+}
+
+function decodeUplink(input) {
+	return {
+	  data : Decoder(input.bytes, input.fPort),
+	};
+}
+
+function normalizeUplink(input) {
+  var data = {};
+  var water = {};
+
+  if (input.data.WATER_LEAK_STATUS) {
+    water.leak = input.data.WATER_LEAK_STATUS ? true : false
```

Comment:
`WATER_LEAK_STATUS` is a string which can be `LEAK`. What value does it have when there's no leak? In any case, a ternary operator is probably not what we want here.

Reply:
Leak status can only be `LEAK` or `NO LEAK` because of this: https://github.com/TheThingsNetwork/lorawan-devices/blob/af5ac51e9203464dfab4a15de02270750b2752d9/vendor/dragino/lwl03a.js#L124

But ive changed it so itll give false for everything that isnt `LEAK` https://github.com/TheThingsNetwork/lorawan-devices/blob/af5ac51e9203464dfab4a15de02270750b2752d9/vendor/dragino/lwl03a.js#L155